### PR TITLE
Improve diagnostics for operating check MODPERMS-175

### PR DESCRIPTION
**Pull request by adamdickmeiss**
Collect all permissions that are failing (rather than report first failure)
Report the modulePermissions as seen from mod-permissions.

### CODE REVIEW

**Review by MikhailFokanov — APPROVED** [review]
Looks good.

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 1]
```diff
@@ -1020,6 +1020,17 @@ static JsonObject getPayloadWithoutValidation(String token) {
     }
   }
 
+  static void checkPermList(String type, List<String> failedPerms, String operatingUser,
+      List<String> modulePermissions) {
+    if (!failedPerms.isEmpty()) {
+      throw new OperatingUserException(
+          "Cannot add " + type + " permission" + (failedPerms.size() > 1 ? "s" : "")
+              + " " + String.join(", ", failedPerms) + " not owned by operating user "
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:1028]:
```suggestion
          "Cannot add " + type + (failedPerms.size() > 1 ? " permissions " : " permission ")
              + String.join(", ", failedPerms) + " not owned by operating user "
```

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 2]
```diff
@@ -1020,6 +1020,17 @@ static JsonObject getPayloadWithoutValidation(String token) {
...
+              + operatingUser + (modulePermissions == null || modulePermissions.isEmpty() ?
+              "" : " modulePermissions: " + String.join(", ", modulePermissions)));
```
Comment (julianladisch) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:1030]:
```suggestion
              "" : ", modulePermissions: " + String.join(", ", modulePermissions)));
```

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Pull request merged**